File: Lab_2/task_1/1.ts
function compressString(s: string): string {
    let compressed = "";
    let count = 1;

    for (let i = 1; i <= s.length; i++) {
        if (i < s.length && s[i] === s[i - 1]) {
            count++;
        } else {
            compressed += s[i - 1] + count;
            count = 1;
        }
    }

    if (compressed.length < s.length) {
        return compressed;
    } else {
        console.log("Сжатие не уменьшило строку. Возвращена исходная строка.");
        return s;
    }
}

function main(): void {
    const s = prompt("Введите строку:");
    if (s !== null) {
        console.log("Результат:", compressString(s));
    }
}

main();

File: Lab_2/task_1/1.rb
def compress_string(s)
    compressed = ""
    count = 1

    (1..s.length).each do |i|
        if i < s.length && s[i] == s[i - 1]
            count += 1
        else
            compressed += "#{s[i - 1]}#{count}"
            count = 1
        end
    end

    if compressed.length < s.length
        compressed
    else
        puts "Сжатие не уменьшило строку. Возвращена исходная строка."
        s
    end
end

print "Введите строку: "
s = gets.chomp
puts "Результат: #{compress_string(s)}"
